File: src/main/java/com/extendedae_plus/common/registry/block/assemblerMatrix/portUpload/BlockUpload.java
package com.extendedae_plus.common.registry.block.assemblerMatrix.portUpload;

import appeng.menu.MenuOpener;
import appeng.menu.locator.MenuLocators;
import com.extendedae_plus.common.init.ModItems;
import com.extendedae_plus.common.init.ModMenuTypes;
import com.glodblock.github.extendedae.common.blocks.matrix.BlockAssemblerMatrixBase;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.block.state.StateDefinition;
import net.minecraft.world.level.block.state.properties.BooleanProperty;

public class BlockUpload extends BlockAssemblerMatrixBase<BlockEntityUpload> {
    public static final BooleanProperty LOCKED = BooleanProperty.create("locked");

    public BlockUpload() {
        this.registerDefaultState(this.defaultBlockState()
                .setValue(LOCKED, false));
    }

    @Override
    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
        super.createBlockStateDefinition(builder);
        builder.add(LOCKED);
    }

    @Override
    public Item getPresentItem() {
        return ModItems.PortUpload.get();
    }

    @Override
    public void openGui(BlockEntityUpload tile, Player player) {
        MenuOpener.open(ModMenuTypes.LabelLinkManageable.get(), player, MenuLocators.forBlockEntity(tile));
    }
}

File: src/main/kotlin/com/extendedae_plus/common/registry/block/assemblerMatrix/portUpload/BlockUpload.kt
package com.extendedae_plus.common.registry.block.assemblerMatrix.portUpload

import appeng.menu.MenuOpener
import appeng.menu.locator.MenuLocators
import com.extendedae_plus.common.init.EAEPItems
import com.extendedae_plus.common.init.EAEPMenuTypes
import com.fish.fishlib.util.extension.invoke
import com.glodblock.github.extendedae.common.blocks.matrix.BlockAssemblerMatrixBase
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.Item
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.block.state.StateDefinition
import net.minecraft.world.level.block.state.properties.BooleanProperty
import net.minecraft.world.level.block.state.properties.Property

class BlockUpload : BlockAssemblerMatrixBase<TileUpload>() {
    init {
        this.registerDefaultState(
            this.stateDefinition.any()
                .setValue(PropertyLocked, false)
        )
    }

    override fun createBlockStateDefinition(builder: StateDefinition.Builder<Block?, BlockState?>) {
        super.createBlockStateDefinition(builder)
        builder.add(PropertyLocked)
    }

    override fun getPresentItem(): Item {
        return EAEPItems.PortUpload.get()
    }

    override fun openGui(tile: TileUpload, player: Player) {
        MenuOpener.open(
            EAEPMenuTypes.LabelLinkManageable(),
            player,
            MenuLocators.forBlockEntity(tile)
        )
    }

    companion object {
        @JvmField
        val PropertyLocked: Property<Boolean> = BooleanProperty.create("locked")
    }
}
